examples/ct/ct_handlers_test.go

```diff
@@ -222,13 +255,195 @@ func TestGetRoots(t *testing.T) {
 	client.AssertExpectations(t)
 }
 
-func loadCertsIntoPoolOrDie(t *testing.T) *PEMCertPool {
+// This uses the fake CA as trusted root and submits a chain of just a leaf which should be rejected
+// because there's no complete path to the root
+func TestAddChainMissingIntermediate(t *testing.T) {
+	km := new(crypto.MockKeyManager)
+	client := new(trillian.MockTrillianLogClient)
+
+	roots := loadCertsIntoPoolOrDie(t, []string{testonly.FakeCACertPem})
+	reqHandlers := CTRequestHandlers{0x42, roots, client, km, time.Millisecond * 500, fakeTimeSource}
+
+	pool := loadCertsIntoPoolOrDie(t, []string{testonly.LeafSignedByFakeIntermediateCertPem})
+	chain := createJsonChain(t, *pool)
+
+	recorder := makeAddChainRequest(t, reqHandlers, chain)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code, "expected HTTP BadRequest for incomplete add-chain: %v", recorder.Body)
+	km.AssertExpectations(t)
+	client.AssertExpectations(t)
+}
+
+// This uses a fake CA as trusted root and submits a chain of just a precert leaf which should be
+// rejected
+func TestAddChainPrecert(t *testing.T) {
+	km := new(crypto.MockKeyManager)
+	client := new(trillian.MockTrillianLogClient)
+
+	roots := loadCertsIntoPoolOrDie(t, []string{testonly.CACertPEM})
+	reqHandlers := CTRequestHandlers{0x42, roots, client, km, time.Millisecond * 500, fakeTimeSource}
+
+	precert, err := fixchain.CertificateFromPEM(testonly.PrecertPEMValid)
+	_, ok := err.(x509.NonFatalErrors)
+
+	if err != nil && !ok {
+		t.Fatal(err)
```

Comment:
As discussed I don't require that there is a non fatal error for the test to succeed. If the CT x509 is modified to accept the precert extension without complaint then I don't want this test to fail.